README.md

```diff
@@ -1,2 +1,243 @@
+  path: string
+  content: string
+  language: 'typescript' | 'json'
+}
+
+interface FileChange {
+  path: string
+  content: string
+  lastModified: Date
+}
+
+interface FileWithChanges {
+  path: string
+  originalContent: string
+  currentContent: string
+  hasChanges: boolean
+  language: 'typescript' | 'json'
+}
+
+interface EditorStore {
+  // Original state from server
+  originalFiles: Record<string, OriginalFile>
+  
+  // User changes
+  changes: Record<string, FileChange>
+  
+  // UI state
+  currentFilePath: string | null
+
+  // Actions
+  updateContent: (path: string, content: string) => void
+  resetChanges: (paths?: string[]) => void
+  setCurrentFile: (path: string) => void
+  
+  // Computed
+  getFilesWithChanges: () => Record<string, FileWithChanges>
+  getCurrentFile: () => FileWithChanges | null
+  initializeEditor: (config: {
+    originalFiles: Record<string, OriginalFile>
+  }) => void
+
+}
+```
+
+### Component Props
+```typescript
+interface TscircuitCodeEditorProps {
+  // Appearance
+  className?: string
+  theme?: 'light' | 'dark'
+  fontSize?: number
+  showLineNumbers?: boolean
+  
+  // Toolbar configuration
+  showToolbar?: boolean
+  toolbarItems?: React.ComponentType
+  
+  // Event handlers
+  onSave?: (file: FileWithChanges) => Promise<void>
+  onError?: (error: Error) => void
+  
+  // Custom components
+  LoadingComponent?: React.ComponentType
+  ErrorComponent?: React.ComponentType<{ error: Error }>
+  FileListComponent?: React.ComponentType<{
+    files: Record<string, FileWithChanges>
+    currentPath: string | null
+    onSelect: (path: string) => void
+  }>
+}
+```
+
+### Migration Guide
+
+1. Install the package:
+```bash
+bun install @tscircuit/code-editor
+```
+
+2. Replace existing CodeEditor implementation with TscircuitCodeEditor:
+```typescript
+// Before
+<CodeEditor
+  onCodeChange={handleCodeChange}
+  readOnly={isReadOnly}
+  initialCode={code}
+  isStreaming={isStreaming}
+  showImportAndFormatButtons={showToolbar}
+/>
+
+// After
+const CircuitEditor = () => {
+  const [isLoading, setIsLoading] = useState(true)
+
+  const { getCurrentFile, initializeEditor } = useTscircuitEditor()
+  
+  // Load files from server
+  useEffect(() => {
+    const loadFiles = async () => {
+      setIsLoading(true)
+      try {
+        const response = await fetch(`/package_files/list?package_release_id=${releaseId}`)
+        const files = await response.json()
+        
+        // Initialize editor with original files
+        initializeEditor({
+          originalFiles: files.reduce((acc, file) => ({
+            ...acc,
+            [file.path]: {
+              path: file.path,
+              content: file.content,
+              language: file.path.endsWith('.json') ? 'json' : 'typescript'
+            }
+          }), {})
+        })
+      } catch (error) {
+        console.error('Failed to load files:', error)
+      }
+      setIsLoading(false)
+    }
+    
+    loadFiles()
+  }, [releaseId])
```

Comment:
add a usage thats kind of similar as an alternate:

```suggestion
  const { getCurrentFile, initializeEditor } = useTscircuitEditor({
    loadInitialFiles: async () => {
      // fetches
      // after return, this will call initializeEditor internall
      return vfs as Record<string, string>
    }
  })
```